File: src/main/java/pl/patrykdolata/lifemanager/config/SecurityProperties.java
package pl.patrykdolata.lifemanager.config;

import lombok.Getter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.web.cors.CorsConfiguration;

@ConfigurationProperties(prefix = "security", ignoreUnknownFields = false)
public class SecurityProperties {

    @Getter
    private final CorsConfiguration cors = new CorsConfiguration();
}

File: src/main/kotlin/pl/patrykdolata/lifemanager/config/SecurityProperties.kt
package pl.patrykdolata.lifemanager.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import org.springframework.web.cors.CorsConfiguration

@ConfigurationProperties(prefix = "security", ignoreInvalidFields = true)
@ConstructorBinding
data class SecurityProperties(val cors: CorsConfiguration)
